core/kernel/tee_ta_manager.c

```diff
@@ -613,13 +616,20 @@ TEE_Result tee_ta_invoke_command(TEE_ErrorOrigin *err,
 	set_invoke_timeout(sess, cancel_req_to);
 	res = sess->ctx->ops->enter_invoke_cmd(sess, cmd, param, err);
 
+	tee_ta_clear_busy(sess->ctx);
+
 	if (sess->ctx->panicked) {
+		/* Clear all other sessions connected to panicked instance */
+		TAILQ_FOREACH(s, open_sessions, link) {
```

Comment:
With this other sessions will not know that the TA which it's connected to has panicked. The client will instead see some strange error, perhaps TEE_ERROR_INVALID_PARAMETER or TEE_ERROR_ITEM_NOT_FOUND. This doesn't seem very friendly to normal world, it could result in normal world not being able to free the non-secure resources allocated for the session.

Reply:
Maybe we should have a `panicked` flag field in `struct tee_session` to properly handle panicked session invocations (invocation return code, session closure through `TA_CloseSessionEntryPoint()`.

Panicked TA context (`struct tee_ta_ctx`) could be fully freed when TA panics, only instances of `struct tee_session` would remain until non-secure closes each sessions.

Reply:
That should work. Perhaps it would be enough to assign NULL to `struct tee_ta_session::ctx`.

Reply:
It is also my first impression (in the [comment](https://github.com/OP-TEE/optee_os/pull/2838#issuecomment-466385647) above where I mentioned something like "will mark the sessions as affected") which could be basically creating an additional flag in 'struct tee_session' to signalize the panic state.
So, in case of panic all the sessions connected will be marked as "panicked", but not closed (any call to those will return 'TEE_ERROR_TARGET_DEAD' until they were closed by the user) and the context will be destroy afterwards. 
In the session closing procedure there will be a check for valid context before destroying it.